opentelemetry-sdk/src/metrics/internal/mod.rs

```diff
@@ -135,6 +134,66 @@ where
             );
         }
     }
+
+    /// Iterate through all attribute sets and populate `DataPoints` in readonly mode.
+    pub(crate) fn collect_readonly<Res, MapFn>(&self, dest: &mut Vec<Res>, mut map_fn: MapFn)
```

Comment:
Another issue worth exploring further: Instead of passing mutable Vec or owned Vec to Exporters, we can simply pass immutable refs, so the sdk can allocate the DataPoints once, and maintain it forever. Exporters just get a immutable ref to Datapoints only. For another PR definitely!

Reply:
I'm not sure if I follow this idea... you mean having immutable `&Vec<DataPoints>`, then how would you update value and time, and exemplars... ? (but maybe let's leave this for another time :))

Reply:
> then how would you update value and time, and exemplars... ?

Exporters don't need to update them, so they don't need mutable one. (lets comeback to this in future)

Reply:
resolving as this is no a blocker for this PR.